### PR TITLE
[Event Hubs Client] Track Two: Second Preview (Documentation Updates)

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The intent of these changes is to update documentation for the library with context related to the Preview Two scope.

# Last Upstream Rebase

Monday, August 5, 2019  11:06am (EDT)

# Resources

- [Azure Messaging Exception Documentation](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-messaging-exceptions)
- [.NET Event Hubs Client: Second Preview Design Proposal](https://github.com/jsquire/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/design/proposal-event-hubs-second-preview.md)

# Related and Follow-Up Issues

- [[Epic] Release Event Hubs Track 2 Library Preview 2](https://github.com/Azure/azure-sdk-for-net/issues/6752) (#6752)
- [Review and update documentation](https://github.com/Azure/azure-sdk-for-net/issues/6795) (#6795)

### CODE REVIEW

**Comment by jsquire** [comment]
//cc: @chrisjbarr

File: packages.md [hunk 1]
```diff
@@ -8,8 +8,7 @@
 | [`Azure.Core`]( sdk/core/Azure.Core/src/Azure.Core.csproj )| [Readme](sdk/core/Azure.Core/README.md) |  N/A  | [Nuget]( https://www.nuget.org/packages/Azure.Core/ ) |
 | [`Azure.Core.Extensions`]( sdk/core/Azure.Core.Extensions/src/Azure.Core.Extensions.csproj )|  N/A  |  N/A  | [Nuget]( https://www.nuget.org/packages/Azure.Core.Extensions/ ) |
 | [`Azure.Identity`]( sdk/identity/Azure.Identity/src/Azure.Identity.csproj )| [Readme](sdk/identity/Azure.Identity/README.md) |  N/A  | [Nuget]( https://www.nuget.org/packages/Azure.Identity/ ) |
-| [`Azure.Messaging.EventHubs`]( sdk/eventhub/Azure.Messaging.EventHubs/src/Azure.Messaging.EventHubs.csproj )| [Readme](sdk/eventhub/Azure.Messaging.EventHubs/README.md) |  N/A  | [Nuget]( https://www.nuget.org/packages/Azure.Messaging.EventHubs/ ) |
-| [`Azure.Messaging.EventHubs.TrackOne`]( sdk/eventhub/Azure.Messaging.EventHubs/src/TrackOneClient/Azure.Messaging.EventHubs.TrackOne.csproj )| [Readme](sdk/eventhub/Azure.Messaging.EventHubs/README.md) |  N/A  |  N/A  |
```
Comment (jsquire) [inline packages.md]: This was never intended as an actual package; it was the result of a mistake during the first preview release.

File: sdk/eventhub/Azure.Messaging.EventHubs/CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,45 @@
...
+
+- Members of type `DateTimeOffset` have had the `Utc` suffix removed from their names, as the `DateTimeOffset` type removes ambiguity about the time zone
+
+- Members representing a partition offset that were incorrectly exposed as `string` have been converted to `int` to provide a unified type.
+
+- Renamed all instances of `EventHubPath` to `EventHubName` to align with the usage context and unify on the chosen semantics across the client library for different languages.
+
+- Ongoing minor tweaks to formatting, wording, and spelling of documentation and XML document comments.
+
+###	Publishing Events
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/CHANGELOG.md]: nit:  Spacing.

File: sdk/eventhub/Azure.Messaging.EventHubs/README.md [hunk 3]
```diff
@@ -115,20 +115,76 @@ await using (var client = new EventHubClient(connectionString, eventHubName))
...
+            // At this point, the loop will wait for events to be available in the partition.  When an event 
+            // is available, the loop will iterate with the received event available.  Because we did not 
+            // specify a maximum wait time, the loop will wait forever unless cancellation is requested using
+            // the cancellation token.
+        }
     }
 }
 ```
 
+### Consume events from all partitions of an Event Hub
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/README.md]: Please consider this a basic skeleton;  @kinelski will revisit in the near future and smooth things out as well as correct any mistakes in the usage that I've made.

File: sdk/eventhub/Azure.Messaging.EventHubs/CHANGELOG.md (hunk 2 above)
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/CHANGELOG.md]: Super weird.  My editor doesn't show these spaces.

Comment (conniey) [reply sdk/eventhub/Azure.Messaging.EventHubs/CHANGELOG.md]: Odd it doesn't show the tab character.

Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/CHANGELOG.md]: Yeah, they didn't act like tabs either, but deleting and recreating worked.   /shrug

File: sdk/eventhub/Azure.Messaging.EventHubs/README.md [hunk 4]
```diff
@@ -115,20 +115,76 @@ await using (var client = new EventHubClient(connectionString, eventHubName))
...
+    await Task.Delay(TimeSpan.FromMinutes(5));
+    await processor.StopAsync();
+}
+```
 ## Troubleshooting
 
 ### Common exceptions
 
-#### Operation Cancelled
+#### Operation Canceled
```
Comment (conniey) [inline sdk/eventhub/Azure.Messaging.EventHubs/README.md]: Sentence casing for headers? Would this be "Operation cancelled"? ;) See what I did there.

File: sdk/eventhub/Azure.Messaging.EventHubs/README.md [hunk 5]
```diff
@@ -115,20 +115,76 @@ await using (var client = new EventHubClient(connectionString, eventHubName))
...
+            // specify a maximum wait time, the loop will wait forever unless cancellation is requested using
+            // the cancellation token.
+        }
     }
 }
 ```
 
+### Consume events from all partitions of an Event Hub
+
+To consume events for all partitions of an Event Hub, you'll create an `EventProcessor` for a specific consumer group.  When an Event Hub is created, it provides a default consumer group that can be used to get started.
```
Comment (conniey) [inline sdk/eventhub/Azure.Messaging.EventHubs/README.md]: nit: trailing spaces

File: sdk/eventhub/Azure.Messaging.EventHubs/CHANGELOG.md [hunk 6]
```diff
@@ -1,4 +1,45 @@
...
+
+- Members of type `DateTimeOffset` have had the `Utc` suffix removed from their names, as the `DateTimeOffset` type removes ambiguity about the time zone
+
+- Members representing a partition offset that were incorrectly exposed as `string` have been converted to `int` to provide a unified type.
+
+- Renamed all instances of `EventHubPath` to `EventHubName` to align with the usage context and unify on the chosen semantics across the client library for different languages.
+
+- Ongoing minor tweaks to formatting, wording, and spelling of documentation and XML document comments.
+
+#### Publishing Events
```
Comment (conniey) [inline sdk/eventhub/Azure.Messaging.EventHubs/CHANGELOG.md]: Consider switching to sentence-style capitalisation for headers. :D
https://docs.microsoft.com/en-ca/style-guide/capitalization

**Review by conniey — APPROVED** [review]
LGTM

File: sdk/eventhub/Azure.Messaging.EventHubs/README.md (hunk 4 above)
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/README.md]: I hate this with every fiber of my being, and came very close to asserting that this is a development artifact and not one that falls under the documentation standards _(which I truly believe.)_    However, you're right and I've changed it.

File: sdk/eventhub/Azure.Messaging.EventHubs/CHANGELOG.md (hunk 6 above)
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/CHANGELOG.md]: see above.   I've done it, but I'm not at all in agreement with the style guide.

File: sdk/eventhub/Azure.Messaging.EventHubs/README.md (hunk 3 above)
Comment (ramya-rao-a) [reply sdk/eventhub/Azure.Messaging.EventHubs/README.md]: How about `Consume events using an Event Processor` instead?

As a user, if I wanted to receive from all partitions, I could still very well use the previous sample and have it run for all partitions.

By having the words `Event Processor` in the header, I hope to draw attention to another way of consuming events that might be of interest to the user

Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/README.md]: Sure.  worked into the samples PR.

**Pull request merged**